src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -134,6 +157,41 @@
                  (inc proj-yr)
                  projection'))))))
 
+(defn apply-national-trend
+  "Takes a dataset of projected rates or values for the jumpoff year, and a dataset of
+   projected national rates for the following years of the projection (currently
+   this is ONS data). Takes parameters for first & last projection years, the
+   scenario to be used from the future rates dataset (e.g. :low, :principal, or 
+   :high), and the name of the column with rates or value in the jumpoff year dataset
+   (e.g. :death-rate or :death). Returns a dataset with projected rates or values
+   for each age, sex, and projection year, calculated by applying the trend from 
+   the projected national rates to the data from the jumpoff year dataset."
+  [jumpoffyr-projection future-assumption
+   first-proj-yr last-proj-yr scenario assumption-col]
+  (let [proportional-differences (calc-proportional-differences future-assumption
+                                                                first-proj-yr
+                                                                last-proj-yr
+                                                                scenario)
+        projection-first-proj-yr (-> jumpoffyr-projection
+                                     (ds/add-column :year (repeat first-proj-yr))
+                                     (ds/select-columns [:gss-code :sex :age :year assumption-col]))]
+    (-> (fn [{:keys [accumulator last-calculated]} current-yr]
```

Comment:
it feels a little strange to me to pass in the function as a first in thread-first.
Usually the idea behind threading imho is to have a pipeline to take the data that you're going to transform and eventually output ... I'd either put it in a let, extract it as a function, or just chuck it into the reduce.